### DISCUSSION TITLE
nvim-tree resizes other windows after opening

### DISCUSSION THREAD

**Discussion by alloc33** [nvim-tree/nvim-tree.lua#1798] [Q&A]
I have an issue with preserving window proportions. If I have one or more horizontal splits - it doesn't work. For example I have terminal in the bottom (:term://zsh or ToggleTerm terminal). So nvim-tree keeps resize my splits igoring 'preserve_window_proportions' property.

Is there any way to fix that?

![ezgif-5-ba467a3353](https://user-images.githubusercontent.com/45129790/205246626-66bbdfa9-902f-46b7-9ae0-e991e1c0783c.gif)

**Comment by alex-courtis** [comment]
Unfortunately `:help nvim-tree.view.preserve_window_proportions` only applies to windows that nvim-tree open, rather than the nvim-tree window itself.

Resizing after split is actually a vim option: `:help 'equalalways'` and out of nvim-tree's control. Try `:set noequalalways`

**Reply by alloc33** [reply]
setting `equalalways=true` unfortunally doesn't help

**Reply by alex-courtis** [reply]
Suggestion: try to replicate the behaviour you desire without nvim-tree.

Create a "fake nvim-tree" window with, say width 30 and try to produce your desired behaviour with manual window opens etc.

**Comment by vatosarmat** [comment]
On `WinNew`
```vim
  set winfixwidth
  set winfixheight
```

**Reply by vatosarmat** [reply]
Or `open_window` in `lua/nvim-tree/view.lua` should be rewritten like this.
```lua
local function open_window()
  if M.View.float.enable then
    vim.api.nvim_open_win(0, true, open_win_config())
  else
    -- vim.api.nvim_command "vsp"
    -- M.reposition_window()
    local width = get_width()
    vim.api.nvim_command(('vert topleft %s split'):format(width))
  end
  setup_tabpage(vim.api.nvim_get_current_tabpage())
  set_window_options_and_buffer()
end
```

**Reply by jsh3lly** [reply]
This works to preserve window sizes when I open nvim-tree with NvimTreeToggle. But the moment I close nvim-tree, the window sizes are wrongly resized to "default" sizes.
